tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/packaging/PathDependencyTestCase.java

```diff
@@ -603,6 +603,27 @@ public void testRuntimeTimeDependencyForExecutingModuleTests() throws BallerinaT
         buildLogLeecher.waitForText(10000);
     }
 
+    /**
+     * Build TestProject in interop-failure. TestProject has utils module which has an interop error.
+     * When there is an interop error, tests should not be executed.
+     *
+     * @throws BallerinaTestException Error when executing the commands.
+     */
+    @Test(description = "Test whether execution of tests are broken when interop failure exists")
+    public void testInteropFailureBreakRunningTests() throws BallerinaTestException {
+        Path caseResources = tempTestResources.resolve("interop-failure");
+        Path execJarPath = Paths.get(TARGET_DIR_NAME, BIN_DIR_NAME, "utils" + BLANG_COMPILED_JAR_EXT);
+
+        String errorMsg = "error: wso2/utils:0.1.0::main.bal:5:1: {ballerina/java}CLASS_NOT_FOUND" +
+                " 'org.wso2.test.StaticMethods'";
+        LogLeecher utilsCompileLeecher = new LogLeecher(errorMsg, LogLeecher.LeecherType.ERROR);
+        balClient.runMain("build", new String[]{"-a"}, envVariables, new String[]{},
+                new LogLeecher[]{utilsCompileLeecher}, caseResources.resolve("TestProject").toString());
+        utilsCompileLeecher.waitForText(5000);
+
+        Assert.assertFalse(Files.exists(execJarPath));
+    }
```

Comment:
If I get it right, this test does not truly assert whether the test execution is failed, as the log leecher is used only to capture the compilation error. How do you ensure that the tests won't be running after the error?